vm/src/obj/objzip.rs

```diff
@@ -19,16 +19,18 @@ impl PyValue for PyZip {
     }
 }
 
-fn zip_new(cls: PyClassRef, iterables: Args, vm: &VirtualMachine) -> PyResult<PyZipRef> {
-    let iterators = iterables
-        .into_iter()
-        .map(|iterable| objiter::get_iter(vm, &iterable))
-        .collect::<Result<Vec<_>, _>>()?;
-    PyZip { iterators }.into_ref_with_type(vm, cls)
-}
-
 #[pyimpl]
 impl PyZip {
+
+    #[pyslot(new)]
+    fn zip_new(cls: PyClassRef, iterables: Args, vm: &VirtualMachine) -> PyResult<PyZipRef> {
```

Comment:
Can this be cleaned up further?  I noticed this note

```
/// `Args` optionally accepts a generic type parameter to allow type checks
/// or conversions of each argument.
```
and that a couple of other classes look look like they do this, but I'm not familiar enough with RustPython yet to construct that code.  I tried it and got stuck at the compile errors.

Reply:
I think it's fine, you could have it be `Args<PyIterable>`, and then add a `iter_obj` method to `PyIterable` to get the object returned from `__iter__`, but I think this is fine if you don't want to do that.

Reply:
I think I'll leave it for now -- but if you think that would be a worthwhile addition, I could make another PR later?

Reply:
Yep, feel free if you'd like to.